src/main/java/org/broadinstitute/consent/http/service/ElectionService.java

```diff
@@ -101,13 +102,29 @@ public Election updateElectionById(Election rec, Integer electionId) {
         }
         Date lastUpdate = new Date();
         electionDAO.updateElectionById(electionId, rec.getStatus(), lastUpdate);
-        if(rec.getArchived() != null && rec.getArchived()) {
-            electionDAO.archiveElectionById(electionId, lastUpdate);
-        }
+        archiveElectionByIdAndStatus(electionId, rec.getStatus());
```

Comment:
This pattern is a bit confusing - it's looks like all elections are being archived at this step, but it's really `archiveElectionIfNewStatusIsArchived` which is confusing... perhaps it should just be:

```
if (isArchiveStatus(rec.getStatus()) {
    electionDAO.archiveElectionById(electionId, update);
}
```

Reply:
Yeah, the method name is confusing. I want to have this in a separate method since we need to call it multiple times ... will rework this a bit.